services-custom/dynamodb-enhanced/src/test/java/software/amazon/awssdk/enhanced/dynamodb/internal/operations/UpdateItemOperationTest.java

```diff
@@ -164,34 +157,24 @@ public void generateRequest_withConditionExpression() {
         FakeItemWithSort item = createUniqueFakeItemWithSort();
         item.setOtherAttribute1("value-1");
         UpdateItemOperation<FakeItemWithSort> updateItemOperation =
-            UpdateItemOperation.create(UpdateItemEnhancedRequest.builder(FakeItemWithSort.class)
-                                                                .item(item)
-                                                                .conditionExpression(CONDITION_EXPRESSION)
-                                                                .build());
-        Map<String, AttributeValue> expectedKey = new HashMap<>();
-        expectedKey.put("id", AttributeValue.builder().s(item.getId()).build());
-        expectedKey.put("sort", AttributeValue.builder().s(item.getSort()).build());
+            UpdateItemOperation.create(requestFakeItemWithSort(item, b -> b.conditionExpression(CONDITION_EXPRESSION)));
+
+        UpdateItemRequest request = updateItemOperation.generateRequest(FakeItemWithSort.getTableSchema(), PRIMARY_CONTEXT, null);
+
+
+        String expectedUpdateExpression = "SET " + OTHER_ATTRIBUTE_1_NAME + " = " + OTHER_ATTRIBUTE_1_VALUE +
```

Comment:
Nit: As soon as we exceed a single concatenation operator, I usually find `String.format(..)` to be more readable.

Reply:
Yeah, didn't love this setup either but again, because we're releasing support of the full range of UpdateExpression, there will be a lot of different types of expression strings so it'll be rectified when those are added.